Update developer's guide and uml diagram (#126)
AY2021S1-CS2113-T13-1/tp: docs/DeveloperGuide.md

The architecture design is given in the diagram above. The main components of NotUS are:

1. `InterfaceManager`: Manages the user input as well as the message output from application.

> 1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The parser then
>  makes sense of the information to the respective commands.

Reviewer: ```suggestion
1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The respective parsers then make sense of the information and calls the respective commands.
```

  <br><em>Figure #</em>
</p>

#### <a id="parserManager"><ins>2.3 ParserManager</ins></a>

> The ParserManager manages the creation of specific parser objects based on the type of the command. The parser then
>  makes sense of the user input and calls the respective commands into action.

Reviewer: ```suggestion
The ParserManager manages the creation of specific parser objects based on the type of command. The parser then makes sense of the user input and calls the respective commands into action.
```

Author: This one is the ide auto ident to next line.

1. Receives the user input message as a whole.

> 1. Interprets the type of command and creates the respective parser.

Reviewer: ```suggestion
1. Interprets the type of command and creates the respective parser for each command.
```

1. The parser then splits the message to identify all the parameters provided.
1. Creates and returns the Command class respectively.
 
<p align="center">
  <img alt="Parser" src="diagrams/out/Parser.png" />
  <br><em>Figure #</em>
</p>

> 💡 Note that the alternate paths in the sequence diagram above is not exhaustive. There is an alternate path for each
>  unique command. As there are too many paths, they are omitted from the diagram. The Command objects in the diagram is
>   used to represent a generic Command object that is created through the Parser. Refer to the next figure
>    for more details.

Reviewer: ```suggestion
💡 Note that the alternate paths in the sequence diagram above are not exhaustive. There is an alternate path for each unique command. As there are too many paths, they are omitted from the diagram. The Command objects in the diagram are used to represent a generic Command object that is created through the Parser. Refer to the next figure for more details.
```

#### <a id="tag"><ins>2.7 Tags</ins></a>

The class diagram below denotes the relationship between the TagManager and the Taggable Objects (Notes and Events).

<p align="center">
   <img alt="Parser" src="diagrams/out/TaggableObject.png"/>
   <br><em>Figure #</em>
 </p>
 

> Notes and Events inherit from the abstract class, TaggableObject and TagManager contains a map of individual unique
>  tags to an arraylist of TaggableObjects. The TagManager handles the creation, deletion as well as the tagging and
>   untagging of tags from notes or events.

Reviewer: ```suggestion
Notes and Events inherit from the abstract class, TaggableObject, and TagManager contains a map of individual unique tags to an ArrayList of TaggableObjects. The TagManager handles the creation, deletion as well as the tagging and untagging of tags from notes or events.
```
